go/vt/discovery/healthcheck.go

```diff
@@ -840,7 +840,7 @@ func (hc *HealthCheckImpl) TabletConnection(ctx context.Context, alias *topodata
 	hc.mu.Lock()
 	thc := hc.healthByAlias[tabletAliasString(topoproto.TabletAliasString(alias))]
 	hc.mu.Unlock()
-	if thc == nil || thc.Conn == nil {
```

Comment:
I removed this because I don't think this particular check makes sense (and it causes the race detector to be unhappy).

`thc.Conn` can become `nil` at any time, including _right after_ this check.

Calling `thc.Connection` a few lines below will return a new connection object that will either be usable (if the vttablet on the other side of the connection can be accessed), or it will return an error when the connection is attempted to be used.

Removing this line seems to have impact on some integration tests and I'll try if I can figure out how to update them accordingly, but as-is this check simply leads to flaky behaviour and can't be trusted.

Reply:
It _seems_ correct to remove this check for the reasons you stated. We should wait and see what your test failures tell you.
One thing to note is that there seem to be exactly two paths to `thc.Connection` which, as you noted, is responsible for initializing the connection object. One is from here, and the other is from `stream`.

Reply:
I went back and looked at the blame for this. Perhaps it is actually protecting us from some other race condition.

Reply:
We may end up recreating a connection to a tablet that we are trying to drop.